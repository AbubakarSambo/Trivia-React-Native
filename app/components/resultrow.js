import React from 'react'
import { Icon } from 'react-native-elements'
import { StyleSheet, View, WebView } from 'react-native'
import PropTypes from 'prop-types'

export const ResultRow = ({ item }) => {
  let correct = item.correct_answer === item.selectedAnswer
  return (
    <View style={correct ? styles.correctRow : styles.wrongRow}>
      <Icon
        raised
        size={15}
        name={correct ? 'check' : 'times'}
        type='font-awesome'
        color='#f50' />
      <WebView source={{ html: item.question }} />
    </View>
  )
}
ResultRow.propTypes = {
  item: PropTypes.object.isRequired
}
const styles = StyleSheet.create({
  wrongRow: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    height: 100,
    backgroundColor: 'red',
    borderRadius: 15,
    marginBottom: 10,
    marginRight: 5
  },
  correctRow: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    width: 'auto',
    height: 100,
    backgroundColor: 'green',
    borderRadius: 15,
    marginBottom: 10,
    marginRight: 5
  }
})
