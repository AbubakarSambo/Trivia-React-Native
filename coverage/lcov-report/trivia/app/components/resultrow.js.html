<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for trivia/app/components/resultrow.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">trivia/app/components</a> resultrow.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Icon } from 'react-native-elements'
import { StyleSheet, View, WebView } from 'react-native'
import PropTypes from 'prop-types';
&nbsp;
export const ResultRow = ({ item }) =&gt; {
    let correct = item.correct_answer === item.selectedAnswer
    return (
        &lt;View style={ correct ? styles.correctRow : styles.wrongRow}&gt;
            &lt;Icon
                raised
                size={15}
                name={correct ? 'check' : 'times'}
                type='font-awesome'
                color='#f50' /&gt;
            &lt;WebView source={{ html: item.question }} /&gt;
        &lt;/View&gt;
    )
}
ResultRow.propTypes = {
    item: PropTypes.object.isRequired,
}
const styles = StyleSheet.create({
    wrongRow: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        height: 100,
        backgroundColor: 'red',
        borderRadius: 15,
        marginBottom: 10,
        marginRight: 5
    },
    correctRow: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        width: 'auto',
        height: 100,
        backgroundColor: 'green',
        borderRadius: 15,
        marginBottom: 10,
        marginRight: 5
    }
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Sep 17 2018 19:07:54 GMT+0100 (WAT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
