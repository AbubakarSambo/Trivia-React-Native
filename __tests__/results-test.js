
import React from 'react'
import renderer from 'react-test-renderer'
import { ResultRow } from '../app/components/resultrow'

it('renders correct row on results page', () => {
  let item = {
    correct_answer: true,
    selectedAnswer: true,
    question: 'is russell westbrook the sickest player ever?'
  }
  const rendered = renderer.create(<ResultRow item={item} />).toJSON()
  expect(rendered).toMatchSnapshot()
})
