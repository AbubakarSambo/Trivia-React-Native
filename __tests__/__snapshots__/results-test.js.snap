// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct row on results page 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": 15,
      "display": "flex",
      "flexDirection": "row",
      "height": 100,
      "marginBottom": 10,
      "marginRight": 5,
      "width": "auto",
    }
  }
>
  <View
    onPress={undefined}
    style={
      Array [
        Object {
          "margin": 7,
        },
        Object {
          "borderRadius": 19,
          "height": 34,
          "width": 34,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    underlayColor="white"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#f50",
            "fontSize": 15,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï€Œ
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={undefined}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={true}
      scrollEnabled={undefined}
      source={
        Object {
          "html": "is russell westbrook the sickest player ever?",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
